### PR TITLE
Web doc

### PR DESCRIPTION
**Pull request by jburel**
This is still work-in-progress
Start cleaning a bit the web doc
 * Dropped Trial
 * Removed Apache ref
cc @hflynn @aleksandra-tarkowska

### CODE REVIEW

**Comment by atarkowska** [comment]
Thanks for cleanup, I will have a look tomorrow. Could we please not drop experimental? We are not even testing advance config

**Comment by atarkowska** [comment]
could you please add benchmark update https://github.com/jburel/ome-documentation/pull/1 ?

**Comment by atarkowska** [comment]
thx, I am upgrading omeroweb-install now.

**Comment by jburel** [comment]
I have a PR already opened against omeroweb
i.e. https://github.com/ome/omeroweb-install/pull/10

**Comment by atarkowska** [comment]
thanks for pointing out. we need a bit more, I will cherry-pick yours and continue

**Comment by jburel** [comment]
@aleksandra-tarkowska thanks I will close my PR in that case

**Comment by jburel** [comment]
@hflynn could you have a look after this first round of cleaning?

**Comment by hflynn** [comment]
Will look today yes

File: omero/sysadmins/troubleshooting.txt [hunk 1]
```diff
@@ -326,8 +326,8 @@ OMERO.web migrating from Apache to Nginx
 
 Since OMERO 5.2.6 support for Apache and mod_wsgi deployment is deprecated.
 It is recommended to use :doc:`/sysadmins/unix/install-web/install-nginx` or
-:doc:`OMERO.web trial deployment </sysadmins/unix/install-web/install-web-trial>`.
-Official support is likely to be dropped during the 5.3.x line.
+:doc:`OMERO.web deployment </sysadmins/unix/install-web/web-deployment>`.
+Official support has been dropped in 5.3.0.
```
Comment (hflynn) [inline omero/sysadmins/troubleshooting.txt]: Rather than just adding this support dropped sentence at the end, I would rephrase the first sentence here:
"Support for Apache and mod_wsgi deployment was deprecated in OMERO 5.2.6 and has been dropped in 5.3.0. It is recommended to use..."

File: omero/sysadmins/unix/install-web.txt [hunk 2]
```diff
@@ -29,9 +29,9 @@ There are a number of good ways to easily deploy OMERO.web.
 
 Deploying OMERO.web separately from OMERO.server is recommended as they
 perform best under different circumstances and require a different set of
-dependencies. Please consider using :doc:`OMERO.web trial deployment <install-web/install-web-trial>`.
+dependencies. Please consider using :doc:`OMERO.web deployment <install-web/web-deployment>`.
```
Comment (hflynn) [inline omero/sysadmins/unix/install-web.txt]: This doc link needs updating to 'OMERO.web WSGI deployment' (or just let it automatically use the page title for the link) because otherwise we have a lot of different 'deployment' sections/titles with no clear differentiation.

In fact, it might also be good to retitle this page as 'Installing OMERO.web' or 'Setting up OMERO.web'

File: omero/sysadmins/unix/install-web.txt [hunk 3]
```diff
@@ -77,6 +73,11 @@ to access the OMERO.webclient:
 OMERO.web maintenance
 ---------------------
 
+If an attempt is made to access OMERO.web whilst
+it is not running, the generated Nginx configuration file will automatically
+display a maintenance page.
+ 
+
```
Comment (hflynn) [inline omero/sysadmins/unix/install-web.txt]: you don't need this extra blank line here

File: omero/sysadmins/unix/install-web.txt [hunk 4]
```diff
@@ -131,17 +132,17 @@ OMERO.web maintenance
...
-
 
 .. _customizing_your_omero_web_installation:
 
 Customizing your OMERO.web installation
 ---------------------------------------
 
+OMERO.web offers a number of configuration options.
+Depending on the deployment choice, OMERO.web will not activate
+configuration changes until either gunicorn is restarted using
```
Comment (hflynn) [inline omero/sysadmins/unix/install-web.txt]: either? Or what? This is confusing. Not clear what '(nginx)' at the end means either

File: omero/sysadmins/unix/install-web/install-nginx.txt [hunk 5]
```diff
@@ -161,92 +160,55 @@ Benchmark
 We run example benchmarks on rendering thumbnails and 512x512 pixels planes
 for 100 concurrent users making 5000 requests in total::
 
-    $ ab -n 5000 -c 100 https://server.openmicroscopy.org/omero/webclient/render_thumbnail/size/96/1234/
-    This is ApacheBench, Version 2.3 <$Revision: 1430300 $>
+    $ ab -n 5000 -c 100 https://server.openmicroscopy.org/omero/webclient/render_thumbnail/1234/
+    This is ApacheBench, Version 2.3 <$Revision: 1706008 $>
```
Comment (hflynn) [inline omero/sysadmins/unix/install-web/install-nginx.txt]: Is this supposed to still be talking about Apache?

File: omero/sysadmins/unix/install-web/install-nginx.txt [hunk 6]
```diff
@@ -282,10 +244,6 @@ using :property:`omero.web.wsgi_args`:
 EXPERIMENTAL: Gunicorn advanced configuration
 ---------------------------------------------
 
-.. note:: Experimental configurations are not ready for production use.
```
Comment (hflynn) [inline omero/sysadmins/unix/install-web/install-nginx.txt]: What is the reasoning behind leaving these subtitles as 'Experimental' but removing the note boxes explaining what that means? Would it be better to reinstate the notes but rephrase them to "these configurations have yet to be extensively tested in a production environment, use at your own risk" or similar?

File: omero/sysadmins/unix/install-web/web-deployment.txt [hunk 7]
```diff
@@ -35,7 +35,7 @@ Prerequisites
    `nginx <http://nginx.org/>`_ and `gunicorn <http://docs.gunicorn.org/>`_,
 
 
-Install the above OMERO.web dependences using the package management tools:
+Install the above OMERO.web dependencies using the package management tools:
```
Comment (hflynn) [inline omero/sysadmins/unix/install-web/web-deployment.txt]: The ordering here is a bit confusing, if the virtualenv bit needs to come before the installation of the dependencies, it might be clearer to have:
"Install the above OMERO.web dependencies using the package management tools (virtualenv is an optional first step):"

File: omero/sysadmins/unix/install-web/web-deployment.txt [hunk 8]
```diff
@@ -311,9 +274,6 @@ For more details refer to
 EXPERIMENTAL: Sync workers
 ^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-.. note:: Experimental configurations are not ready for production use.
```
Comment (hflynn) [inline omero/sysadmins/unix/install-web/web-deployment.txt:314]: As above, not sure about leaving the 'Experimental' title but removing the note about what that means.

**Review by hflynn — COMMENTED** [review]
Definitely getting there but a bunch of suggestions and some minor nitpicks

File: omero/sysadmins/unix/install-web/install-nginx.txt (hunk 5 above)
Comment (jburel) [reply omero/sysadmins/unix/install-web/install-nginx.txt]: this is different
This is used for measurement performance of web servers
The web server does not have to be Apache.

File: omero/sysadmins/unix/install-web/install-nginx.txt (hunk 6 above)
Comment (jburel) [reply omero/sysadmins/unix/install-web/install-nginx.txt]: I initially removed the EXPERIMENTAL but rolled back the changes and forgot to roll back the notes

File: omero/sysadmins/unix/install-web/web-deployment.txt (hunk 8 above)
Comment (jburel) [inline omero/sysadmins/unix/install-web/web-deployment.txt]: we probably do not need 3 note boxes in such a short space.

Comment (jburel) [reply omero/sysadmins/unix/install-web/web-deployment.txt:314]: I have added one note box
I don't think we need 3

**Comment by jburel** [comment]
@hflynn: comments should be addressed

File: omero/sysadmins/unix/install-web/install-nginx.txt (hunk 5 above)
Comment (hflynn) [reply omero/sysadmins/unix/install-web/install-nginx.txt]: 👍 just thought it was worth checking

File: omero/sysadmins/server-upgrade.txt [hunk 9]
```diff
@@ -373,8 +368,8 @@ OMERO.web migrating from Apache to Nginx
 
 Since OMERO 5.2.6 support for Apache and mod_wsgi deployment is deprecated.
 It is recommended to use :doc:`/sysadmins/unix/install-web/install-nginx` or
-:doc:`OMERO.web trial deployment </sysadmins/unix/install-web/install-web-trial>`.
-Official support is likely to be dropped during the 5.3.x line.
+:doc:`OMERO.web deployment </sysadmins/unix/install-web/web-deployment>`.
```
Comment (hflynn) [inline omero/sysadmins/server-upgrade.txt]: This link should be updated to use the page title i.e. remove the `OMERO.web deployment` and the `<>`

File: omero/sysadmins/troubleshooting.txt [hunk 10]
```diff
@@ -324,10 +324,10 @@ mod_wsgi will deadlock if run in daemon mode while mod_python is enabled
 OMERO.web migrating from Apache to Nginx
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Since OMERO 5.2.6 support for Apache and mod_wsgi deployment is deprecated.
-It is recommended to use :doc:`/sysadmins/unix/install-web/install-nginx` or
-:doc:`OMERO.web trial deployment </sysadmins/unix/install-web/install-web-trial>`.
-Official support is likely to be dropped during the 5.3.x line.
+Support for Apache and mod_wsgi deployment was deprecated
+in OMERO 5.2.6. It is recommended to use
```
Comment (hflynn) [inline omero/sysadmins/troubleshooting.txt]: "...was deprecated in OMERO 5.2.6 and dropped in 5.3.0."

File: omero/sysadmins/troubleshooting.txt [hunk 11]
```diff
@@ -324,10 +324,10 @@ mod_wsgi will deadlock if run in daemon mode while mod_python is enabled
...
+:doc:`/sysadmins/unix/install-web/install-nginx` or
+:doc:`OMERO.web WSGI deployment </sysadmins/unix/install-web/web-deployment>`.
```
Comment (hflynn) [inline omero/sysadmins/troubleshooting.txt]: Better to just remove the set link text and let it automatically use the page title

File: omero/sysadmins/unix/install-web.txt [hunk 12]
```diff
@@ -29,9 +29,9 @@ There are a number of good ways to easily deploy OMERO.web.
 
 Deploying OMERO.web separately from OMERO.server is recommended as they
 perform best under different circumstances and require a different set of
-dependencies. Please consider using :doc:`OMERO.web trial deployment <install-web/install-web-trial>`.
+dependencies. Please consider using :doc:`OMERO.web WSGI deployment <install-web/web-deployment>`.
```
Comment (hflynn) [inline omero/sysadmins/unix/install-web.txt]: Again, let's remove the hard coded link text and let the build automatically use the page title

File: omero/sysadmins/unix/install-web/install-nginx.txt [hunk 13]
```diff
@@ -160,92 +160,55 @@ Benchmark
...
-    Time per request:       44.898 [ms] (mean, across all concurrent requests)
-    Transfer rate:          45.46 [Kbytes/sec] received
-
-    Connection Times (ms)
-                  min  mean[+/-sd] median   max
-    Connect:        0    0   0.3      0       3
-    Processing:   435 4446 685.2   4363    7644
-    Waiting:      432 4446 685.3   4362    7644
-    Total:        435 4446 685.1   4363    7644
+    Benchmarking idr-testing.openmicroscopy.org (be patient)
```
Comment (atarkowska) [inline omero/sysadmins/unix/install-web/install-nginx.txt]: could you please replace with `server.openmicroscopy.org` sorry, missed it

File: omero/sysadmins/unix/install-web/web-deployment.txt [hunk 14]
```diff
@@ -1,5 +1,5 @@
-TRIAL: OMERO.web WSGI deployment
-================================
+OMERO.web WSGI deployment
```
Comment (hflynn) [inline omero/sysadmins/unix/install-web/web-deployment.txt:1]: I'm still concerned that the distinction between this page and the deployment with Nginx and Gunicorn page isn't clear, can we add something about how this is for deploying **separately** to OMERO.server to the title? "Deploying OMERO.web separately to OMERO.server using WSGI"? or something similar?

File: omero/sysadmins/unix/install-web/web-deployment.txt (hunk 8 above)
Comment (hflynn) [reply omero/sysadmins/unix/install-web/web-deployment.txt:314]: Yes, that's fine, I agree multiple ones on subheadings is unnecessary

File: omero/sysadmins/unix/install-web/web-deployment.txt [hunk 15]
```diff
@@ -52,8 +52,8 @@ Virtualenv is an optional step:
    2016-08-24 16:20:14,252 [omego.artifa] INFO  Checking http://downloads.openmicroscopy.org/omero/5.2.5/./artifacts/OMERO.py-5.3.0-ice36-b29.zip
```
Comment (hflynn) [inline omero/sysadmins/unix/install-web/web-deployment.txt]: You've still got 5.2.5 as the version number for the artifacts folder here

Comment (hflynn) [reply omero/sysadmins/unix/install-web/web-deployment.txt]: And in the line below

File: omero/sysadmins/unix/install-web/web-deployment.txt [hunk 16]
```diff
@@ -52,8 +52,8 @@ Virtualenv is an optional step:
...
    2016-08-24 16:20:14,252 [omego.fileut] INFO  Downloading http://downloads.openmicroscopy.org/omero/5.2.5/./artifacts/OMERO.py-5.3.0-ice36-b29.zip
    2016-08-24 16:20:14,514 [omego.artifa] INFO  Unzipping OMERO.py-5.3.0-ice36-b29.zip
-  $ ln -s `pwd`/OMERO.py-5.2.5-ice36-b28 `pwd`/OMERO.py
-  $ pip install -r OMERO.py/share/web/requirements-py27-trial.txt
+   $ ln -s `pwd`/OMERO.py-5.2.5-ice36-b28 `pwd`/OMERO.py
```
Comment (hflynn) [inline omero/sysadmins/unix/install-web/web-deployment.txt]: 5.2.5 here too

File: omero/sysadmins/server-upgrade.txt [hunk 17]
```diff
@@ -373,8 +368,8 @@ OMERO.web migrating from Apache to Nginx
...
+Official support has been dropped in 5.3.0.
```
Comment (hflynn) [inline omero/sysadmins/server-upgrade.txt]: Move this to the first sentence please:
"Support for Apache and mod_wsgi deployment was deprecated in OMERO 5.2.6 and has been dropped in 5.3.0."

File: omero/sysadmins/unix/install-web/web-deployment.txt [hunk 18]
```diff
@@ -294,8 +257,8 @@ using :property:`omero.web.wsgi_args`:
 EXPERIMENTAL: Gunicorn advanced configuration
 ---------------------------------------------
 
-.. note:: Experimental configurations are not ready for production use.
-    Configuration may change. Features may not work properly.
+.. note:: The following configuration options have yet to be extensively tested in a
+          production environment, use at your own risk.
```
Comment (hflynn) [inline omero/sysadmins/unix/install-web/web-deployment.txt]: Unnecessary extra whitespace here, you don't need to indent the 2nd line this far, just to underneath the 'o' of note

**Review by hflynn — CHANGES_REQUESTED** [review]
A bunch more nitpicks and clarification between the different pages still needs some work I think

File: omero/sysadmins/unix/install-web/install-nginx.txt [hunk 19]
```diff
@@ -161,92 +160,55 @@ Benchmark
...
-      50%   4363
-      66%   4553
-      75%   4670
-      80%   4750
-      90%   5072
-      95%   5398
-      98%   6795
-      99%   6955
-     100%   7644 (longest request)
+    Benchmarking idr-testing.openmicroscopy.org (be patient)
```
Comment (hflynn) [inline omero/sysadmins/unix/install-web/install-nginx.txt]: idr mention here too

File: omero/sysadmins/unix/install-web/web-deployment.txt [hunk 20]
```diff
@@ -177,92 +177,55 @@ Benchmark
...
-      66%   4553
-      75%   4670
-      80%   4750
-      90%   5072
-      95%   5398
-      98%   6795
-      99%   6955
-     100%   7644 (longest request)
-
+    Benchmarking idr-testing.openmicroscopy.org (be patient)
```
Comment (hflynn) [inline omero/sysadmins/unix/install-web/web-deployment.txt]: idr mention

File: omero/sysadmins/unix/install-web/web-deployment.txt (hunk 15 above)
Comment (jburel) [reply omero/sysadmins/unix/install-web/web-deployment.txt]: for that I will see when we have the new walkthrough in so we can use litteralinclude

File: omero/sysadmins/unix/install-web/web-deployment.txt (hunk 16 above)
Comment (jburel) [reply omero/sysadmins/unix/install-web/web-deployment.txt]: same

**Comment by jburel** [comment]
@hflynn done further cleaning as mentioned.
Hopefully this should be a bit clearer

File: omero/sysadmins/unix/install-web.txt [hunk 21]
```diff
@@ -16,37 +16,41 @@ with:
...
     :maxdepth: 1
     :titlesonly:
 
-    install-web/install-nginx
-    install-web/web-deployment
     install-web/walkthrough/omeroweb-install-centos7-ice3.6
     install-web/walkthrough/omeroweb-install-ubuntu-ice3.6
     install-web/walkthrough/omeroweb-install-osx-ice3.6
 
+.. toctree::
```
Comment (hflynn) [inline omero/sysadmins/unix/install-web.txt]: Can we move this above the walkthroughs so the page is the next one as you scroll through please?

File: omero/sysadmins/unix/install-web.txt [hunk 22]
```diff
@@ -16,37 +16,41 @@ with:
...
 perform best under different circumstances and require a different set of
-dependencies. Please consider using :doc:`OMERO.web WSGI deployment <install-web/web-deployment>`.
+dependencies. See :doc:`install-web/web-deployment` for more details.
 
 If you have installed Nginx, OMERO can automatically generate a
 configuration file for your web server. The location of the file will depend
 on your system, please refer to your web server's manual. See
 :ref:`customizing_your_omero_web_installation` for additional
-customization options.
+customization options. Below is a list of OS specific walkthroughs.
```
Comment (hflynn) [inline omero/sysadmins/unix/install-web.txt]: I think it would probably be clearer is the 'Below is a list...' sentence was a new paragraph (below the hidden toctree) and maybe an extra few words of explanation would be helpful

File: omero/sysadmins/unix/install-web/web-deployment.txt [hunk 23]
```diff
@@ -290,12 +256,14 @@ using :property:`omero.web.wsgi_args`:
 
     $ OMERO.py/bin/omero config set omero.web.wsgi_args -- "--log-level=DEBUG --error-logfile=/home/omero/OMERO.server/var/log/error.log".
 
+.. note:: The following configuration options marked as **EXPERIMENTAL** have yet to be extensively tested in a
```
Comment (hflynn) [inline omero/sysadmins/unix/install-web/web-deployment.txt]: This note should be below the heading

**Comment by hflynn** [comment]
2 more nitpicks but this is otherwise looking good now thanks

**Review by hflynn — APPROVED** [review]

**Comment by hflynn** [comment]
I'll check this over with fresh eyes tomorrow but I think it should be sorted now thanks

File: omero/sysadmins/unix/install-web/web-deployment.txt [hunk 24]
```diff
@@ -1,15 +1,13 @@
-TRIAL: OMERO.web WSGI deployment
-================================
+OMERO.web deployment
```
Comment (hflynn) [inline omero/sysadmins/unix/install-web/web-deployment.txt]: Sorry but can we restore the 'WSGI' in the title here to distinguish this from the deployment subsections elsewhere?

**Comment by hflynn** [comment]
@jburel one last change to that page title and then this is good to merge!

**Comment by hflynn** [comment]
👍
